src/web_sdfv.ts

```diff
@@ -0,0 +1,1180 @@
+    if (param)
+        return parseScriptParamValue(param);
+
+    return def;
+}
+
+$(() => {
+    // Do not run initiailization code if inside of VSCode.
+    try {
+        // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+        vscode;
+        if (vscode)
+            return;
+    } catch (_) { }
+
+    // Set the default settings based on the current script's attributes
+    // or URL parameters.
+    const isEmbedded = document.getElementById('embedded') !== null;
+    if (isEmbedded) {
+        SDFVSettings.set<boolean>('toolbar', false);
+        SDFVSettings.set<boolean>('minimap', false);
+    }
+
+    // Check if any of the remaining settings are provided via the URL.
+    for (const key of SDFVSettings.settingsKeys) {
+        const overrideVal = settingReadDefault(key, undefined);
+        if (overrideVal !== undefined)
+            SDFVSettings.set(key, overrideVal);
+    }
+
+    WebSDFV.getInstance().init();
+
+    // If the renderer is not null, an SDFG has already been set from somewhere
+    // else.
+    if (WebSDFV.getInstance().renderer === undefined) {
+        if (document.currentScript?.hasAttribute('data-sdfg-json')) {
+            const sdfgString =
+                document.currentScript.getAttribute('data-sdfg-json');
+            if (sdfgString) {
+                void WebSDFV.getInstance().setSDFG(
+                    checkCompatLoad(parseSDFG(sdfgString)), null
+                );
+            }
+        } else if (document.currentScript?.hasAttribute('data-url')) {
+            const url =
+                document.currentScript.getAttribute('data-url');
+            if (url)
+                loadSDFGFromURL(url);
+        } else {
+            const url = getParameterByName('url');
+            if (url)
+                loadSDFGFromURL(url);
+            else
+                void WebSDFV.getInstance().setSDFG(null);
+        }
+    }
+});
+
+// Define global exports outside of webpack
+declare global {
+    interface Window {
+        // Extensible classes for rendering and overlays
+        OverlayManager: typeof OverlayManager;
+        GenericSdfgOverlay: typeof GenericSdfgOverlay;
+        SDFGElement: typeof SDFGElement;
+
+        // API classes
+        SDFV: typeof SDFV;
+        WebSDFV: typeof WebSDFV;
+        SDFGRenderer: typeof SDFGRenderer;
+
+        // Exported functions
+        parseSDFG: (sdfgJson: string) => JsonSDFG;
+        stringifySDFG: (sdfg: JsonSDFG) => string;
+        checkCompatLoad: (sdfg: JsonSDFG) => JsonSDFG;
+        checkCompatSave: (sdfg: JsonSDFG) => JsonSDFG;
+    }
+}
+
+window.OverlayManager = OverlayManager;
+window.GenericSdfgOverlay = GenericSdfgOverlay;
+window.SDFGElement = SDFGElement;
+window.SDFV = SDFV;
+window.WebSDFV = WebSDFV;
+window.SDFGRenderer = SDFGRenderer;
+window.parseSDFG = parseSDFG;
+window.stringifySDFG = stringifySDFG;
+window.checkCompatLoad = checkCompatLoad;
+window.checkCompatSave = checkCompatSave;
```

Comment:
do we still need all of that? we are now fully typescript/npm-based so I am wondering.